packages/config-loader/src/lib/reader.ts

```diff
@@ -67,9 +67,9 @@ export async function readConfigFile(
     const out: JsonObject = {};
 
     for (const [key, value] of Object.entries(obj)) {
-      const result = await transform(value, `${path}.${key}`);
-      if (result !== undefined) {
-        out[key] = result;
+      // undefined covers optional fields, and nulls are explicitly skipped
```

Comment:
Oh hold up was this to address https://github.com/spotify/backstage/pull/1367#issuecomment-645918480? Because that requires null values to stick around until we're reading out the config

Reply:
No it was just to appease the typechecker. Without it, it complained that undefined wasn't a valid type

Reply:
👌